agb/src/interrupt.rs

```diff
@@ -21,37 +28,344 @@ pub enum Interrupt {
 const ENABLED_INTERRUPTS: MemoryMapped<u16> = unsafe { MemoryMapped::new(0x04000200) };
 const INTERRUPTS_ENABLED: MemoryMapped<u16> = unsafe { MemoryMapped::new(0x04000208) };
 
-pub fn enable(interrupt: Interrupt) {
+fn enable(interrupt: Interrupt) {
     let _interrupt_token = temporary_interrupt_disable();
+    other_things_to_enable_interrupt(interrupt);
     let interrupt = interrupt as usize;
     let enabled = ENABLED_INTERRUPTS.get() | (1 << (interrupt as u16));
     ENABLED_INTERRUPTS.set(enabled);
 }
 
-pub fn disable(interrupt: Interrupt) {
+fn disable(interrupt: Interrupt) {
     let _interrupt_token = temporary_interrupt_disable();
+    other_things_to_disable_interrupt(interrupt);
     let interrupt = interrupt as usize;
     let enabled = ENABLED_INTERRUPTS.get() & !(1 << (interrupt as u16));
     ENABLED_INTERRUPTS.set(enabled);
 }
 
-pub struct Disable {}
+fn other_things_to_enable_interrupt(interrupt: Interrupt) {
```

Comment:
these can also be part of the `impl Interrput` to keep things a bit more self contained